### PR TITLE
Fixed array form of merge which is variadic.

### PR DESCRIPTION
**Pull request by AlexMouton**

### CODE REVIEW

File: src/Rxjs.res [hunk 1]
```diff
@@ -128,7 +128,7 @@ external toObservable: t<'c, 's, 'a> => t<foreign, void, 'a> = "%identity"
 let const: ('b, . t<'c, 's, 'a>) => t<'c, 's, 'b> = (b) => map(. (_, _) => b)
 
 
-@module("rxjs") external mergeArray: array<t<'ca, 'sa, 'a>> => t<foreign, void, 'a> = "merge"
+@module("rxjs") @variadic external mergeArray: array<t<'ca, 'sa, 'a>> => t<foreign, void, 'a> = "merge"
 @module("rxjs") external merge2: (t<'ca, 'sa, 'a>, t<'cb, 'sb, 'a>) => t<foreign, void, 'a> = "merge"
```
Comment (KateChanthakaew) [inline src/Rxjs.res:132]: Ah I see. reading Reacript docs : looks like `@variadic` is for binding res with JS functions that take an arbitrary amount of arguments. ReScript supports modeling those, under the condition that the arbitrary arguments part is homogenous (aka of the same type).

**Review by KateChanthakaew — APPROVED** [review]

File: src/Rxjs.res (hunk 1 above)
Comment (cid8600) [reply src/Rxjs.res:132]: @AlexMouton Is this a fix that helps us get further along in FieldTripv1 integration or is this a general RxJS enhancement? Mind just writing your reasoning on the update in the comment below? Our plan is to merge this in, update our main repo with the new package version, and then run smokes /nightlies against the upgrade. If everything works as expected, we'll deploy this fix into our main production application. 

At that point, where would we stand on FieldTripv1 issues? Does this unblock getting to the bottom of those? Just hoping to get a coordinated road map to help further this along. Cheers!

**Comment by AlexMouton** [comment]
Hey Cid

Yeah I need this for fields.   FieldArray merges an array of observables
and it was producing the wrong thing.
Rescript library versions are a pain in that the final project only
compiles against one version of the dependency, so you need to
revision everything in step.

A

**Review by cid8600 — APPROVED** [review]

**Comment by AlexMouton** [comment]
Hey Cid I may have discouraged you...can you publish the new version of this so i can use it in my fields latest?

**Comment by AlexMouton** [comment]
Are there some other changes in here since last release besides this @variadic? I am seeing a bunch of changes but forget what happened...

**Pull request merged**